doc/builders/packages/xorg: Convert to CommonMark.

MostAwesomeDude opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Fulfilling #105265.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

MostAwesomeDude committed:
doc/builders/packages/xorg: Convert to CommonMark.

I had to pick 7.4 or 7.5. I picked 7.4 for now.

MostAwesomeDude committed:
doc/builders/packages/xorg: Acknowledge the retired katamari.

I tested the shell snippets shown here; they work. Note that, because
they're intended for copy-and-paste, I did *not* make them into
ShellSessions with prompts. But that's something I can do if desired.

MostAwesomeDude committed:
doc/builders/packages/xorg: Fix pasto.

I did double-check that the fixed version does what we would want. In
either case, the katamari script is mostly of historical value, and
there's nothing to change in pkgs.

ryantm commented on doc/builders/packages/xorg.section.md:
````diff
@@ -0,0 +1,34 @@
+
+The generator is invoked as follows:
+
+```sh
````
```suggestion
```ShellSession
```

ryantm commented on doc/builders/packages/xorg.section.md:
````diff
@@ -0,0 +1,34 @@
+
+X.org upstream releases used to include [katamari](https://en.wiktionary.org/wiki/%E3%81%8B%E3%81%9F%E3%81%BE%E3%82%8A) releases, which included a holistic recommended version for each tarball, up until 7.7. To create a list of tarballs in a katamari release:
+
+```sh
````
```suggestion
```ShellSession
```

MostAwesomeDude committed:
doc/builders/packages/xorg: Use ShellSession for shell.

I was worried, but the HTML output can still be copy-and-pasted, so it's
not a problem.

MostAwesomeDude commented:
Addressed review feedback and also rebased to fix merge conflicts. Thanks for your hard work.
